File: 739-daily-temperatures/739-daily-temperatures.go
func dailyTemperatures(temperatures []int) []int {
  ans := make([]int, len(temperatures))
  stack := make([]int, 0, len(temperatures))
  for i, v := range temperatures {
    for l:=len(stack); l > 0 && v > temperatures[stack[l-1]]; l-- {
      j := stack[l-1]
      ans[j] = i - j
      stack = stack[:l-1]
    }
    stack = append(stack, i)
  }
  return ans
}

File: 739-daily-temperatures/739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
  const ans = Array(temperatures.length).fill(0)
  const stack = []
  temperatures.forEach((v, i) => {
    while (stack.length > 0 && v > temperatures[stack.at(-1)]) {
      const j = stack.pop()
      ans[j] = i - j
    }
    stack.push(i)
  })
  return ans
};